tests/syscalls.rs

```diff
@@ -25,16 +28,30 @@ fn test_contract(
     contract_path: impl AsRef<Path>,
     entry_point: &str,
     class_hash: [u8; 32],
-    nonce: usize,
     contract_address: Address,
     caller_address: Address,
     general_config: StarknetGeneralConfig,
+    tx_context: Option<TransactionExecutionContext>,
     return_data: impl Into<Vec<Felt>>,
 ) {
     let contract_class = ContractClass::try_from(contract_path.as_ref().to_path_buf())
         .expect("Could not load contract from JSON");
 
-    let contract_state = ContractState::new(class_hash, nonce.into(), Default::default());
+    let tx_execution_context = tx_context.unwrap_or_else(|| {
+        TransactionExecutionContext::create_for_testing(
```

Comment:
maybe it is unnecessary to use create_for_testing, we could simple use `new()` and delete boiler plate code.

Reply:
i thought so, and if I remember correctly there's at least one more method that's not really needed.